src/reporting/reportConversionResults.ts

```diff
@@ -93,3 +94,31 @@ const logMissingPlugins = (plugins: Set<string>, logger: Logger) => {
             .join(""),
     );
 };
+
+interface RuleWithNotices {
+    notices: any[];
+    ruleName: string;
+}
+
+const logNotices = (converted: Map<string, ESLintRuleOptions>, logger: Logger) => {
+    const rulesWithNotices = Array.from(converted.values()).filter(
+        ruleOptions => ruleOptions.notices && ruleOptions.notices.length >= 1,
+    ) as RuleWithNotices[];
```

Comment:
@JoshuaKGoldberg is this kind of what you had in mind? (Essentially casting each rule within the new array)

Reply:
Yes, pretty much 🚀 thanks!

Reply:
As a post-merge FYI - the way I was thinking of was:

```ts
const rulesWithNotices = Array.from(converted.values()).filter(
    (ruleOptions): ruleOptions is RuleWithNotices =>
        !!(ruleOptions.notices && ruleOptions.notices.length >= 1)
);
```

...which IMO is _slightly_ preferable because using `as` casts is kind of a way around the type system (rather than informing it of the stuff we already know). But this is such a small difference that it doesn't seem reasonable to block the PR on it.